app/src/main/java/com/example/una/fragments/QuickDonateFragment.java

```diff
@@ -79,56 +86,91 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         bottomSheetBehavior = BottomSheetBehavior.from(bottomSheet);
 
-        client.getFavoriteCharity(new OnSuccessListener<DocumentSnapshot>() {
+        tvAmountDonated.setText("-");
+        tvTimesDonated.setText("-");
+        tvNumChallegesCreated.setText("-");
+
+        populateData();
+        etCustomAmount.addTextChangedListener(new CurrencyTextWatcher(etCustomAmount, "$0.00"));
+        setValueCheckedListener();
+    }
+
+    private void populateData() {
+        client.getUserDocument(new OnSuccessListener<DocumentSnapshot>() {
             @Override
             public void onSuccess(DocumentSnapshot documentSnapshot) {
                 String charityName = (String) documentSnapshot.get("fav_charity_name");
                 String charityEin;
 
+                boolean isValid = true;
+
+                if (documentSnapshot.contains("num_challenges")) {
+                    numChallenges = (long) documentSnapshot.get("num_challenges");
+                } else {
+                    isValid = false;
+                    numChallenges = 0;
+                }
+
+                if (documentSnapshot.contains("num_donations")) {
+                    numDonations = (long) documentSnapshot.get("num_donations");
+                } else {
+                    isValid = false;
+                    numDonations = 0;
+                }
+
+                if (documentSnapshot.contains("donation_total")) {
+                    totalAmountDonated = (Double) documentSnapshot.get("donation_total");
+                } else {
+                    isValid = false;
+                    totalAmountDonated = 0.0;
+                }
+
+                tvNumChallegesCreated.setText(numChallenges + "");
+                tvAmountDonated.setText("$" + df.format(totalAmountDonated));
+                tvTimesDonated.setText(numDonations + "");
+
+                if (!isValid) {
+                    client.updateUserStreaks(numDonations, numChallenges, totalAmountDonated,
+                            new OnSuccessListener() {
+                                @Override
+                                public void onSuccess(Object o) {
+                                    Log.i(TAG, "updated user streaks");
+                                }
+                            }, new OnFailureListener() {
+                                @Override
+                                public void onFailure(@NonNull Exception e) {
+                                    Log.i(TAG, "failed to update user streaks");
+                                }
+                            });
+                }
+
                 if (documentSnapshot.contains("fav_charity_ein")) {
                     charityEin = documentSnapshot.get("fav_charity_ein").toString();
                 } else {
                     charityEin = getString(R.string.red_cross_ein);
                 }
 
+                if (charityName == null) {
+                    charityName = "";
```

Comment:
Does this occur? Maybe you can set the visibility of tvCharityName to be gone if appropriate